update_ip/ip_getters/base.py

```diff
@@ -1,42 +1,49 @@
 import re
-import urllib2
 import socket
+import urllib2
 
 socket.setdefaulttimeout(5)
 
+
 class GetIpFailed(Exception):
     pass
 
-ip_regex= re.compile("\d{1,3}\.\d{1,3}\.\d{1,3}\.\d{1,3}")
+
+# Check for valid IP Addresses
+ip_regex = re.compile(r"(?:(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\.){3}(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)")
```

Comment:
Thanks for the tip! I know that I can evaluate specific instances of text against it - but I wouldn't know *which* specific instances I should test. I can try a few randomly, but it seems very likely my tests would not be comprehensive.

>if need more explanation this web can explain each part.
I can certainly understand each part - it's just that this takes too much time to understand as a whole, and is way too complex for what it does. I had to literally paste it into the python interpreter to make sure the second part is equal to the first.

In any case, we're going to have to match for ipv6 addresses pretty soon. I'm definitely not going to [do that with a regex](https://stackoverflow.com/questions/53497/regular-expression-that-matches-valid-ipv6-addresses#17871737) ;)
I don't understand the whole.